### PR TITLE
Server service actor

### PR DESCRIPTION
**Pull request by jonyrock**
## Motivation

There is still a problem that we can't guarantee than analytics is alive if main thread alive.
This should fix https://github.com/hastic/hastic-server/issues/316 and https://github.com/hastic/hastic-server/issues/407 in theory, but load test needed to check

## Implementation 

`ServerService` now is `AsyncZmqActor` so it works is a separate thread.

### CODE REVIEW

File: analytics/analytics/services/server_service.py [hunk 1]
```diff
@@ -39,60 +42,75 @@ def fromJSON(json: dict):
...
+    async def _on_message_to_thread(self, message: str):
+        await self.__server_socket.send_string(message)
+
+    async def __server_socket_recv_loop(self):
+        while True:
+            received_string = await self.__server_socket.recv_string()
+            if received_string == 'PING':
+                asyncio.ensure_future(self.__handle_ping())
+            else:
+                self._send_message_from_thread(received_string)
```
Comment (rozetko) [inline analytics/analytics/services/server_service.py]: Why don't we wrap it in `asyncio.ensure_future`?

File: analytics/analytics/services/server_service.py [hunk 2]
```diff
@@ -39,60 +42,75 @@ def fromJSON(json: dict):
...
 
     async def __handle_ping(self):
-        await self.socket.send(b'PONG')
+        await self.__server_socket.send(b'PONG')
```
Comment (rozetko) [inline analytics/analytics/services/server_service.py]: Maybe we can do `send_string('PONG')` here?

File: analytics/analytics/services/server_service.py [hunk 3]
```diff
@@ -39,61 +46,79 @@ def fromJSON(json: dict):
...
+        # self.__actor_socket.send_string(json.dumps(message.toJSON()))
+        await self._put_message_to_thread(json.dumps(message.toJSON()))
+
+    async def send_request_to_server(self, message: ServerMessage) -> object:
+        if message.request_id is not None:
+            raise ValueError('Message can`t have request_id before it is scheduled')
+        request_id = message.request_id = self.__request_next_id
+        self.request_next_id = self.__request_next_id + 1
+        asyncio.ensure_future(self.send_message_to_server(message))
+        while request_id not in self.__responses:
```
Comment (jonyrock) [inline analytics/analytics/services/server_service.py]: you should `await self.__responses[request_id]` which should be a task, which you resolve somewhere else

Comment (jonyrock) [inline analytics/analytics/services/server_service.py]:
```suggestion
        # you should await self.__responses[request_id] which should be a task, 
        # which you resolve somewhere else
        while request_id not in self.__responses:
```

**Pull request merged**